silx/test/utils.py

```diff
@@ -243,6 +239,57 @@ def temp_dir():
         shutil.rmtree(tmp_dir)
 
 
+# Simulate missing library context
+class EnsureImportError(object):
+    """This context manager allows to simulate the unavailability
+    of a library, even if it is actually available. It ensures that
+    an ImportError is raised if the code inside the context tries to
+    import the module.
+
+    It can be used to test that a correct fallback library is used,
+    or that the expected error code is returned.
+
+    Trivial example::
+
+        from silx.test.utils import EnsureImportError
+
+        with EnsureImportError("h5py"):
+            try:
+                import h5py
+            except ImportError:
+                print("Good")
+
+    .. note::
+
+        This context manager does not remove the library from the namespace,
+        if it is already imported. It only ensures that any attempt to import
+        it again will cause an ImportError to be raised.
+    """
+    def __init__(self, name):
+        """
+
+        :param str name: Name of module to be hidden (e.g. "h5py")
+        """
+        self.module_name = name
+
+    def __enter__(self):
+        """Simulate failed import by setting sys.modules[name]=None"""
```

Comment:
You can simplify the `if else` with:
`self._backup = sys.modules.get(self.module_name)`

and use `if self._backup is None:` in `__exit__`.

Reply:
I thought there could be a case where sys.modules[self.module_name] exists and is is already None, if someone already tried to import and it genuinely failed. Better not to delete it in such a case?
.

Reply:
OK, I didn't considered this case.